server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/plan/node/write/InsertRowsOfOneDeviceNode.java

```diff
@@ -153,7 +153,9 @@ public void validateAndSetSchema(ISchemaTree schemaTree)
         this.failedMeasurementIndex2Info = insertRowNode.failedMeasurementIndex2Info;
       }
     }
-    storeMeasurementsAndDataType();
+    if (!this.hasFailedMeasurements()) {
+      storeMeasurementsAndDataType();
```

Comment:
I don't know why it is called here.

Reply:
If we don't fill datatypes in AnalyzeVisitor, the datatypes should be stored here?